velero-plugins/imagestream/shared.go

```diff
@@ -78,5 +102,64 @@ func findSpecTag(tags []imagev1API.TagReference, name string) *imagev1API.TagRef
 	return nil
 }
 
+func getRoute(namespace string, location string, configMap string) (string, error) {
+
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return "Error in cluster config", err
+	}
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return "Error in getting client", err
+	}
+	cMap := client.CoreV1().ConfigMaps(namespace)
+	mapClient, err := cMap.Get(configMap, metav1.GetOptions{})
+	if err != nil {
+		return "Error in getting config map", err
+	}
+	osClient, err := routev1client.NewForConfig(config)
+	if err != nil {
+		return "Error in getting client for route", err
+	}
+	routeClient := osClient.Routes(namespace)
+	route, err := routeClient.Get(mapClient.Data[location], metav1.GetOptions{})
+	if err != nil {
+		return "No route found", err
+	}
+	return route.Spec.Host, nil
+}
+
+func getBackup(name string, namespace string) (string, error) {
+	config, err := rest.InClusterConfig()
+	crdConfig := *config
+	crdConfig.ContentConfig.GroupVersion = &schema.GroupVersion{Group: "velero.io", Version: "v1"}
+	crdConfig.APIPath = "/apis"
+	crdConfig.NegotiatedSerializer = serializer.NewCodecFactory(scheme.Scheme)
+	crdConfig.UserAgent = rest.DefaultKubernetesUserAgent()
+	result := velero.BackupList{}
+
+	if err != nil {
+		panic(err)
+	}
+	client, err := rest.UnversionedRESTClientFor(&crdConfig)
+	if err != nil {
+		return "", err
+	}
 
+	err = client.
+		Get().
+		Namespace(namespace).
+		Resource("backups").
+		Do().
+		Into(&result)
+	if err != nil {
+		return "", err
+	}
+	for _, element := range result.Items{
+		if element.Name == name{
+			return element.Spec.StorageLocation, nil
+		}
+	}
+	return "default", nil
```

Comment:
So this is returning `default` since that's the BSL we assume exists if nothing is specified. +1